### PR TITLE
Allow nullptr as data argument to constructor

### PR DESCRIPTION
**Pull request by WardBrian**
This was requested in #103.

Currently, you can specify a model has no data by passing "". This also allows you to pass `nullptr` for the same thing.

### CODE REVIEW

File: src/bridgestan.cpp [hunk 1]
```diff
@@ -9,14 +9,14 @@ int bs_patch_version = BRIDGESTAN_PATCH;
...
+bs_model_rng* bs_construct(const char* data, unsigned int seed,
                            unsigned int chain_id, char** error_msg) {
   try {
-    return new bs_model_rng(data_file, seed, chain_id);
+    return new bs_model_rng(data, seed, chain_id);
   } catch (const std::exception& e) {
     if (error_msg) {
       std::stringstream error;
-      error << "construct(" << data_file << ", " << seed << ", " << chain_id
+      error << "construct(" << data << ", " << seed << ", " << chain_id
```
Comment (aseyboldt) [inline src/bridgestan.cpp:19]: I guess this is invalid if `data` is `NULL`?

Comment (WardBrian) [reply src/bridgestan.cpp:19]: It seems that you can use `operator<<` with a null pointer but it terminates the string. E.g, you get an error which looks like:
`Error: construct(` \<EOL\>
which isn't great. 

It's easy to special-case, though.

Comment (aseyboldt) [reply src/bridgestan.cpp]: Sorry for all the pedantic comments recently.
Seems I get that way when I write too much rust. :-)

Comment (WardBrian) [reply src/bridgestan.cpp]: Absolutely no need to apologize, I appreciate them. Every pedantic comment now is a future bug report -> pr -> patch release cycle we get to skip.

(coincidentally, this is why I like languages like Rust or OCaml with picky compilers. The downside is sometimes it ends up with me being far too trusting of my C++ compiler...)

**Comment by bob-carpenter** [comment]
@aseyboldt:  You picked the right team for pedantry.  We really do appreciate all the detailed comments.

File: src/bridgestan.h [hunk 2]
```diff
@@ -18,9 +18,9 @@ extern int bs_patch_version;
  * generator (PRNG) wrapper.  Data must be encoded in JSON as
  * indicated in the *CmdStan Reference Manual*.
  *
- * @param[in] data_file C-style string. This is either a
- * path to JSON-encoded data file (must end with ".json"), or
- * a JSON string literal.
+ * @param[in] data C-style string. This is either a
+ * path to JSON-encoded data file (must end with ".json"),
+ * a JSON string literal, or nullptr, which is interpreted as no data.
```
Comment (roualdes) [inline src/bridgestan.h:23]:
```suggestion
 * a JSON string literal, an empty string, or nullptr, which is interpreted as no data.
```

**Review by roualdes — APPROVED** [review]
Just one doc suggestions.  Otherwise, looks good to me.

File: src/bridgestan.h (hunk 2 above)
Comment (WardBrian) [reply src/bridgestan.h]: Agreed, it is good to call out this case specifically.

**Pull request merged**